### ISSUE TITLE
import.meta.env cannot be used??

### ISSUE THREAD

**Issue by nyoung697** [szymmis/vite-express#11]
I am trying to setup some env. variables following this guide: https://vitejs.dev/guide/env-and-mode.html

I get the error: `[The 'import.meta' meta-property is only allowed when the '--module' option is 'es2020', 'esnext', or 'system'.ts(1343)](https://stackoverflow.com/questions/69716191/the-import-meta-meta-property-is-only-allowed-when-the-module-option-is-e)`

I noticed that if I scaffold a project using `npm create vite`, it sets the module in tsconfig to `ESNext`, however vite-express sets it to `CommonJS`. I get build errors on the server side if I change this back to ESNext.

Is there a recommended way to get vite to inject environment variables into the frontend during the build?

**Comment by szymmis** [comment]
Hi @nyoung697, thanks a lot for pointing out this issue. Have you tried running the app despite the error? I've tested it and it seems that app works correctly and `import.meta.env` is loaded by Vite as you would expect. The problem occurs only in vscode or other code editor you are using and it is quite easy to fix. I'll include the patch in next version of `create-vite-package` but you can use the fix right now as you've already scaffolded the project. You need to add a separate `tsconfig.json` that will have `module` set to `esnext`. It's the best idea to put it inside `src/client` directory as vscode should pick it up automatically.

It can look something like this 

```json
{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "module": "ESNext"
  } 
}
```

This example works for the basic configuration create by `create-vite-express`, you can adjust it to your needs, but the important part is the `"module": ESNext"` as it makes vscode stop complaining about `import.meta`

You don't need to change anything else because it breaks the app as you pointed out. The only thing left to remember is that you might have to reload TS Server inside vscode by ussing the command palette or by manually restarting the editor.

If you have any questions feel free to ask, I'll try to answer the best i can!

**Comment by nyoung697** [comment]
Great! Thank you for your reply @szymmis.

**Comment by szymmis** [comment]
Should be fixed in create-vite-express v0.0.5.
